Add spec helper for CLLocationManager auth status

joemasilotti opened the pull request:
This PR opens up the class method `+authorizationStatus` on `CLLocationManager` with `+setAuthorizationStatus:`.

It uses the same method swizzling used in Foundation and UIKit subpods and follows the same convention of copying over the code as UIKit does. This should be refactored into a shared location, perhaps having UIKit and CoreLocation depend on Foundation? I'm not 100% sure how that works with sub pod specs and any thoughts are appreciated.

akitchen commented:
This build is red due to a compilation problem, can you have a look @joemasilotti  ?

joemasilotti commented:
Turns out I had added the helper to the wrong target. I fixed it and pushed a new version.

Now the build is failing due to the following issue:

> Test target Foundation-StaticLibSpecBundle encountered an error (Unable to run app in Simulator. An error was encountered while running (Domain = FBSOpenApplicationErrorDomain, Code = 4))

I've now seen a few build fail with this message over the past week or so. Any idea what's going on here? A quick SO search led to people suggesting to just reset the simulator.

akitchen commented:
Your fork is missing 170 commits from master - could you try replaying this commit on top of a new branch based on pivotal/master ?

joemasilotti commented:
OK everything is rebased on top of the most recent master.

Unfortunately I can't get the build to pass on CI. Running `rake` locally works 100% of the time locally even when fresh cloning the repo.

The build on CI alternates between an ambiguous compile issue or the one that I mentioned earlier. Any ideas on how to fix this?

joemasilotti commented:
Turns out I was importing the wrong Cedar header in the spec. @idoru and I tracked it down today and I'm pushing an update now.

akitchen commented:
Great!  Could you also update the CHANGES.markdown ?

joemasilotti commented:
Sure thing. I just updated the file.

akitchen commented:
Thanks Joe!
